src/Paprika/Data/SlottedArray.cs

```diff
@@ -236,7 +239,16 @@ public void MoveNonEmptyKeysTo(in MapSource destination, bool treatEmptyAsTombst
             Span<byte> data;
 
             NibblePath trimmed;
-            if (slot.HasKeyBytes)
+
+            int len = slot.KeyPreamble >> 1;
+            bool isOdd = (slot.KeyPreamble & 1) != 0;
+
+            if (len is >= 5 and <= 6)
```

Comment:
Two `if` for comparing length. For cases where there len is 0, 1, 2, 3 it will require 3 comparisons. Why not to check 

```
if (len >= 5)
{
  if (len <= 6)
  {
  }
  else
  {
  }
}
```

or other stuff like that 😛

Reply:
Done :)
Looks much cleaner now